R/ebirdregioninfo.R

```diff
@@ -49,15 +50,34 @@ ebirdregioninfo <- function(loc, format = "full", key = NULL, ...) {
   tt <- GET(URLencode(url), 
             query = ebird_compact(args), 
             add_headers("X-eBirdApiToken" = get_key(key)), ...)
+  
+  if (tt$status_code == 410) {
+    stop(paste("No such hotspot.",loc))
+  }
   stop_for_status(tt)
   
   ss <- content(tt, as = "text", encoding = "UTF-8")
   json <- jsonlite::fromJSON(ss, FALSE)
   if (length(json) == 0) stop(paste("No region with code", loc))
   
   df <- as_data_frame(t(rapply(json, unlist))) 
-  # rename column name for region info (defaults to "result")
-  if (names(df)[1] == "result") names(df)[1] <- "region" 
-  
+  # rename column name for region info (defaults to "result") 
+  if (names(df)[1] == "result") {
+    names(df)[1] <- "region" 
+  } 
+  # set numeric and logical types correctly
+  if (regtype== "region") {
```

Comment:
I hadn't picked up that all columns were being returned as character! This is a good starting point, but I'll edit it so the fixing of column classes is more robust to changes in the column order (which could easily happen in the API side).